sdk/credentials.go

```diff
@@ -98,6 +98,22 @@ func (c *credentials) GetOrganization(ctx context.Context) (string, error) {
 	return c.organization, nil
 }
 
+// GetUser first ensures a token has been parsed before returning the user,
+// as it is extracted from the token claims.
+// credentials.m2mProfile.User should not be accessed directly, but rather through this method.
```

Comment:
To make it more generic, as we either access m2m or agent profiles:

```suggestion
// credentials.<profile>.User should not be accessed directly, but rather through this method.
```